java-openliberty/samples/jpa/src/main/java/org/example/app/health/DatabaseReadinessCheck.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2020 IBM Corporation and others
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.example.app.health;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import javax.annotation.Resource;
+import javax.enterprise.context.ApplicationScoped;
+import javax.sql.DataSource;
+import org.eclipse.microprofile.health.Readiness;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+
+/**
+ * Determines the database's readiness to process application requests.
+ */
+@Readiness
+@ApplicationScoped
+public class DatabaseReadinessCheck implements HealthCheck {
+
+    @Resource
+    private DataSource datasource;
+
+    @Override
+    public HealthCheckResponse call() {
+        try {
+            Connection connection = datasource.getConnection();
+            DatabaseMetaData connData = connection.getMetaData();
+            boolean valid = connection.isValid(30);
```

Comment:
Why '30' ?   From here:  https://github.com/OpenLiberty/application-stack/blob/master/dev/app-deploy.yaml  we have:

>    periodSeconds: 2
>    timeoutSeconds: 1

Not an expert here but wonder if '1' would be better...to match?

Reply:
Changed it to 1. This is a sample. No need to try to guess what the user's optimal value would be because we can't. Users will most surely update this value to one that fits the state of their cluster.